src/test/java/teammates/test/cases/webapi/GetFeedbackSessionsActionTest.java

```diff
@@ -20,16 +32,280 @@ protected String getRequestMethod() {
+        String[] submissionParam = {
+                Const.ParamsNames.IS_IN_RECYCLE_BIN, "false",
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+        };
+
+        GetFeedbackSessionsAction action = getAction(submissionParam);
+        FeedbackSessionsData fsData = (FeedbackSessionsData) getJsonResult(action).getOutput();
+
+        assertEquals(5, fsData.getFeedbackSessions().size());
+        for (FeedbackSessionData sessionData : fsData.getFeedbackSessions()) {
+            assertEquals(instructor2OfCourse1.getCourseId(), sessionData.getCourseId());
+            assertNull(sessionData.getDeletedAtTimestamp());
+            assertTrue(sessionsInCourse1.stream().anyMatch(session
+                    -> session.getFeedbackSessionName().equals(sessionData.getFeedbackSessionName())));
+        }
+    }
+
+    @Test
+    protected void testExecute_instructorAsStudent_shouldReturnAllSessionsForStudent() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        StudentAttributes student2InCourse2 = typicalBundle.students.get("student2InCourse2");
+
+        loginAsStudentInstructor(instructor1OfCourse1.googleId);
+        String[] submissionParam = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+
+        GetFeedbackSessionsAction action = getAction(submissionParam);
+        FeedbackSessionsData fsData = (FeedbackSessionsData) getJsonResult(action).getOutput();
+
+        assertEquals(2, fsData.getFeedbackSessions().size());
+        for (FeedbackSessionData sessionData : fsData.getFeedbackSessions()) {
+            assertEquals(student2InCourse2.getCourse(), sessionData.getCourseId());
+            assertNull(sessionData.getDeletedAtTimestamp());
+            assertTrue(sessionsInCourse2.stream().anyMatch(session
+                    -> session.getFeedbackSessionName().equals(sessionData.getFeedbackSessionName())));
+        }
+    }
+
+    @Test
+    protected void testExecute_instructorAsStudentWithCourseId_shouldReturnAllSessionsForCourseOfStudent() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        StudentAttributes student2InCourse2 = typicalBundle.students.get("student2InCourse2");
+
+        loginAsStudentInstructor(instructor1OfCourse1.googleId);
+
+        String[] submissionParam = {
+                Const.ParamsNames.COURSE_ID, student2InCourse2.getCourse(),
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+
+        GetFeedbackSessionsAction action = getAction(submissionParam);
+        FeedbackSessionsData fsData = (FeedbackSessionsData) getJsonResult(action).getOutput();
+
+        assertEquals(2, fsData.getFeedbackSessions().size());
+        for (FeedbackSessionData sessionData : fsData.getFeedbackSessions()) {
+            assertEquals(student2InCourse2.getCourse(), sessionData.getCourseId());
```

Comment:
You should have at least one test which verify the full attributes returned by the API endpoint.